sys/userlib/src/critical_section.rs

```diff
@@ -0,0 +1,29 @@
+//! An implementation to support the `critical-section` crate within a Hubris
+//! user task.
+//!
+//! Hubris is careful never to introduce non-local or asynchronous control flow
+//! into a program, and doesn't have threads. This means that, within the
+//! context of a task, we don't actually need to generate any code to implement
+//! a critical section -- they happen naturally.
+//!
+//! You might want to opt out of this implementation if you're doing something
+//! very weird with shared memory. By default, it is hard to do that in Hubris,
+//! but it is possible, and we'll assume you know what you've signed yourself up
+//! for should you choose to do it.
```

Comment:
should this documentation explain _how_ you would opt out of this `critical_section` implementation if you're doing something weird? or is the idea that not telling users how to do that will discourage them from doing it? :)

Reply:
I am undecided tbh; it's a matter of disabling the `critical-section` feature on `userlib`, but I think the likelihood that we'll ever have to do it is pretty low.

Reply:
yeah, i think it's probably obvious enough that you don't need to explain it, especially if we build the docs with the Rustdoc feature that shows required feature flags...